tests_integration/test_nativeapp.py

```diff
@@ -564,3 +565,264 @@ def test_nativeapp_init_deploy(
                 env=TEST_ENV,
             )
             assert result.exit_code == 0
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "command,contains,not_contains",
+    [
+        # deploy --prune removes remote-only files
+        ["app deploy --prune", ["stage/README.md"], ["stage/manifest.yml"]],
+        # deploy removes remote-only files (--prune is the default value)
+        ["app deploy", ["stage/README.md"], ["stage/manifest.yml"]],
+        # deploy --no-prune does not delete remote-only files
+        ["app deploy --no-prune", ["stage/manifest.yml"], []],
+    ],
+)
+def test_nativeapp_deploy_prune(
+    command,
+    contains,
+    not_contains,
+    runner,
+    snowflake_session,
+    temporary_working_directory,
+):
+    project_name = "myapp"
+    result = runner.invoke_json(
+        ["app", "init", project_name],
+        env=TEST_ENV,
+    )
+    assert result.exit_code == 0
+
+    with pushd(Path(os.getcwd(), project_name)):
+        result = runner.invoke_with_connection_json(
+            ["app", "deploy"],
+            env=TEST_ENV,
+        )
+        assert result.exit_code == 0
+
+        try:
+            # delete a file locally
+            os.remove(os.path.join("app", "manifest.yml"))
```

Comment:
nit: consider deleting a different file, the fact that we don't check for a manifest now is somewhat of a bug.